host/sgx/linux/sgxquoteexloader.c

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <dlfcn.h>
+#include <openenclave/internal/defs.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/trace.h>
+#include <stdlib.h>
+#include <string.h>
+#include "../../../common/oe_host_stdlib.h"
+#include "../sgxquote_ex.h"
+
+static const char* _quote_ex_library_file_name = "libsgx_quote_ex.so";
+
+static oe_sgx_quote_ex_library_t* _quote_ex_library = NULL;
+
+static void _unload_quote_ex_library(void)
+{
+    if (_quote_ex_library && _quote_ex_library->handle)
+    {
+        OE_TRACE_INFO(
+            "_unload_quote_ex_library() %s\n", _quote_ex_library_file_name);
+        dlclose(_quote_ex_library->handle);
+        _quote_ex_library->handle = 0;
+        if (_quote_ex_library->mapped)
+        {
+            oe_free(_quote_ex_library->mapped);
+            _quote_ex_library->mapped = NULL;
+        }
+        if (_quote_ex_library->uuid)
+        {
+            oe_free(_quote_ex_library->uuid);
+            _quote_ex_library->uuid = NULL;
+        }
+        if (_quote_ex_library->sgx_key_id)
+        {
+            oe_free(_quote_ex_library->sgx_key_id);
+            _quote_ex_library->sgx_key_id = NULL;
+        }
+    }
+}
+
+void oe_sgx_load_quote_ex_library(oe_sgx_quote_ex_library_t* library)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    if (!library)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    _quote_ex_library = library;
```

Comment:
Since a global variable `_quote_ex_library` is being set, it is not clear if we have made this code thread-safe.

Reply:
@anakrish the top-level function that set and manipulate this variable is `_load_quote_ex_library_once()` in `host/sgx/sgxquote.c`. This function is only run once, as guarded by a mutex in function `_use_quote_ex_library()`.

Reply:
A better place for the comment is in the header file that declares this this function that requires platform specific implementation? I'll add comments there (`host/sgx/sgxquote_ex.h`)